static/app/utils/withSentryRouter.tsx

```diff
@@ -0,0 +1,28 @@
+// eslint-disable-next-line no-restricted-imports
+import {withRouter, WithRouterProps} from 'react-router';
+
+import {customerDomain, usingCustomerDomain} from 'sentry/constants';
+
+/**
+ * withSentryRouter is a higher-order component (HOC) that wraps withRouter, and implicitly injects the current customer
+ * domain as the orgId parameter. This only happens if a customer domain is currently being used.
+ *
+ * Since withRouter() is discouraged from being used on new React components, we would use withSentryRouter() on
+ * pre-existing React components.
```

Comment:
Should we also add a lint error to prevent this HoC from getting a foothold in new code? Ideally new code is built using functional components and `useParams()` but life always finds a way :smile:

Reply:
@markstory I created an eslint lint rule to discourage the use of this higher-order component in https://github.com/getsentry/eslint-config-sentry/pull/166.